students/scotchwsplenda/Lesson_5/MailRoom_3.py

```diff
@@ -26,80 +25,87 @@ def main():
-            "\n"+"1 - Print list of extant donors"
-            "\n"+"2 - Enter donor name and donation"
-            "\n"+"3 - Return to the MAILROOM main menu"
-            "\n"+"Indicate your choice:  ")
-        if respondy == "1":
-            for k in donators:
-                print(k)
-        if respondy == "2":
-            donor_inp = input("Input donor: ")
-            if donor_inp in donators.keys():
-                while True:
-                    try:
-                        don_amount = input("That's a known donor,"
-                                           " input donation amount: ")
-                        don_amount = float(don_amount)
-                        donators[donor_inp].append(don_amount)
-                        send_thanks(donor_inp, don_amount)
-                        break
-                    except ValueError:
-                        print('You have entered a non-number where a number'
-                              ' clearly goes, please get your shit together')
-                        continue
-            else:
-                while True:
-                    try:
-                        new_don = input(
-                            f"{donor_inp}'s an unknown donor, input donation"
-                            " amount: ")
-                        new_don = float(new_don)
-                        donators.update({donor_inp: [new_don]})
-                        send_thanks(donor_inp, new_don)
-                        break
-                    except ValueError:
-                        print('You have entered a non-number where a number'
-                              ' clearly goes, please get your shit together')
-                        continue
-
-        elif respondy == "3":
-            main()
+    print('''
+Please select from the below Thank You Note options:
+1 - Print list of extant donors
+2 - Enter donor name and donation
+3 - Return to the MAILROOM main menu
+    ''')
+    while True:
+        try:
+            respondy = int(input("Indicate your choice:"))
+            if respondy == 1:
+                for k in donators:
+                    print(k)
+                continue
+            if respondy == 2:
+                donor_inp = input("Input donor: ")
+                if donor_inp in donators.keys():
+                    don_amount = input("That's a known donor,"
+                                       " input donation amount: ")
+                    don_amount = float(don_amount)
+                    donators[donor_inp].append(don_amount)
+                    send_thanks(donor_inp, don_amount)
+                else:
+                    new_don = input(
+                        f"{donor_inp}'s an unknown donor, input donation"
+                        " amount: ")
+                    new_don = float(new_don)
+                    donators.update({donor_inp: [new_don]})
+                    send_thanks(donor_inp, new_don)
+            if respondy == 3:
+                mainy()
```

Comment:
as I mentioned before, there should be no separate calls to main() --- you just need to make sure your main loop is a while True loop.